File: algorithms/go/fibonacci-iterative.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"time"
)

const MOD = 1_000_000_007

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n int64
	fmt.Fscan(reader, &n)

	start := time.Now()

	var result int64
	if n == 0 {
		result = 0
	} else {
		a, b := int64(0), int64(1)
		for i := int64(2); i <= n; i++ {
			c := (a + b) % MOD
			a = b; b = c
		}
		result = b
	}

	ms := float64(time.Since(start).Nanoseconds()) / 1e6

	fmt.Fprintln(writer, result)
	fmt.Fprintf(os.Stderr, "%.3f\n", ms)
}

File: algorithms/c/fibonacci-iterative.c
#include <stdio.h>
#include <time.h>

#define MOD 1000000007

int main() {
    long long n;
    scanf("%lld", &n);

    struct timespec start, end;
    clock_gettime(CLOCK_MONOTONIC, &start);

    long long a = 0, b = 1;
    for (long long i = 2; i <= n; i++) {
        long long c = (a + b) % MOD;
        a = b; b = c;
    }
    long long result = (n == 0) ? 0 : b;

    clock_gettime(CLOCK_MONOTONIC, &end);
    double ms = (end.tv_sec  - start.tv_sec)  * 1000.0
              + (end.tv_nsec - start.tv_nsec) / 1e6;

    printf("%lld\n", result);
    fprintf(stderr, "%.3f\n", ms);
    return 0;
}